micrometer-core/src/main/java/io/micrometer/core/instrument/Gauge.java

```diff
@@ -46,10 +49,13 @@ default Type type() {
         private final T obj;
         private final ToDoubleFunction<T> f;
         private final List<Tag> tags = new ArrayList<>();
-        private String description;
-        private String baseUnit;
+        private @Nullable String description;
+        private @Nullable String baseUnit;
 
         private Builder(String name, T obj, ToDoubleFunction<T> f) {
+            Assert.notNull(name,"name");
+            Assert.notNull(obj,"gauge object");
```

Comment:
This change is causing the dropwizard null gauge test to fail. Should `null`s be allowed for `Gauge`s @jkschneider ?

Reply:
It is incorrect. The obj can become dereferenced and, since our gauge implementations maintain a weak reference to it, thus `null`. I think we want to maintain that behavior.

In general, I'm not keen on adding assert code anywhere in the main source set.

Reply:
> I'm not keen on adding assert code anywhere in the main source set.

Sorry I thought you had requested that I add the asserts. I'm fine removing them. Or by `main source set` are you saying to leave them out of the common code paths but leave them in other places like binders?

Reply:
> should I begin adding in NonNull assertions in the constructor?

Sorry, I thought you meant adding `@NonNull` annotations on constructor parameters.